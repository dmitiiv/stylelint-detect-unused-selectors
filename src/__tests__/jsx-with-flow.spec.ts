import path from 'path';
import stylelint from 'stylelint';
import { parseResult } from './__helpers__/parse-result';

const fixturesRoot = path.join(
  __dirname,
  '..',
  '..',
  'examples',
  'jsx-with-flow',
);
const configFilePath = path.join(fixturesRoot, '.stylelintrc.js');

test('Disabllow selectors that are not used in a flow-typed JSX file', async (): Promise<
  void
> => {
  const options = {
    configFile: configFilePath,
    files: path.join(fixturesRoot, '*.css'),
  };

  const result = await stylelint.lint(options);

  expect(result.errored).toBe(true);
  expect(parseResult(result)).toMatchSnapshot();
});
