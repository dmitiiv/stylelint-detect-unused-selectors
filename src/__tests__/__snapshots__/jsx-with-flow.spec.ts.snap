// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disabllow selectors that are not used in a flow-typed JSX file 1`] = `
Object {
  "deprecations": Array [],
  "errored": true,
  "invalidOptionWarnings": Array [],
  "parseErrors": Array [],
  "warnings": Array [
    Object {
      "column": 1,
      "line": 22,
      "rule": "plugin/no-unused-selectors",
      "severity": "error",
      "text": ".not-found is defined but doesn't match any elements in test.jsx. (plugin/no-unused-selectors)",
    },
  ],
}
`;
